cilium-cli/clustermesh/clustermesh.go

```diff
@@ -1301,7 +1304,40 @@ func (k *K8sClusterMesh) connectLocalWithHelm(ctx context.Context, localClient *
 	return k.helmUpgrade(ctx, localClient, state.localHelmValues)
 }
 
+func convertToListClusterMeshValues(values map[string]any) map[string]any {
+	c, found, _ := unstructured.NestedFieldNoCopy(values, "clustermesh", "config", "clusters")
+	if !found || c == nil {
+		c = map[string]any{}
+	}
+	clusterList := []any{}
+	clusterMap := c.(map[string]any)
+	for clusterName, clusterRaw := range clusterMap {
+		cluster := clusterRaw.(map[string]any)
+		cluster["name"] = clusterName
+		clusterList = append(clusterList, cluster)
+	}
+
+	enabledRaw, found, _ := unstructured.NestedFieldNoCopy(values, "clustermesh", "config", "enabled")
+	if !found || enabledRaw == nil {
+		enabledRaw = false
+	}
+	enabled := enabledRaw.(bool)
+
+	return map[string]any{
+		"clustermesh": map[string]any{
+			"config": map[string]any{
+				"enabled":  enabled,
+				"clusters": clusterList,
+			},
+		},
+	}
+}
+
 func (k *K8sClusterMesh) helmUpgrade(ctx context.Context, client *k8s.Client, values map[string]any) error {
+	// FIXME: force output to list for now. This can be removed conditionally
+	//        with a version check once we would be able to set ReuseValues to
+	//        false
+	values = convertToListClusterMeshValues(values)
```

Comment:
Are you planning to address this limitation?

Reply:
Developers are notorious for leaving landmines in the codebase for future developers to find and not following up. If we are OK with the current implementation, I would suggest just deleting the fixme. It's great if you plan to follow up to improve. What I disagree with is documenting such intentions into the tree: If it's important, we should agree it's important and block this PR until it's resolved. If it's not important, then there's no need to leave a note for our future selves. We can agree the current implementation is fine and then just follow up at your discretion.

Reply:
Above is not a hard rule, but general guidance. If something is OK but could be improved, we have from time to time added context in the code with a comment like `// This does X and could be improved by changing to do Y. See GH-XXXX for more details` and file an issue to follow up.